### PR TITLE
Verbose option and dependent parts

### LINKED ISSUE #1193: Add "verbose" option as a config option
Would be easier for some people and would be preferable for the grunt plugin

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/10052707-add-verbose-option-as-a-config-option?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by markelog**
* Add "verbose" option as a config option

* Do not show rule for "Unsupported rule" error 

* Deprecate params for "Checker" constructor

Fixes #1193

### CODE REVIEW

**Comment by mrjoelkemp** [comment]
Update readme to remove it as a cli option? 

This is possibly considered a breaking change if folks rely on the output of the CLI using verbose in flag form.

**Comment by markelog** [comment]
> Update readme to remove it as a cli option?

You can still override it though console, need to update docs though

File: lib/cli.js [hunk 1]
```diff
@@ -19,10 +19,7 @@ module.exports = function(program) {
     var checkerPromise;
     var defer = Vow.defer();
     var promise = defer.promise();
-    var checker = new Checker({
-        verbose: program.verbose,
-        esnext: program.esnext
```
Comment (zxqfox) [inline lib/cli.js]: Unused esnext?

File: lib/errors.js [hunk 2]
```diff
@@ -72,6 +72,20 @@ Errors.prototype = {
     },
 
     /**
+     * Prepare error message.
+     *
+     * @param {Object} errorInfo
+     * @private
+     */
+    _prepareMessage: function(errorInfo) {
+        if (this._verbose && this._currentRule) {
```
Comment (zxqfox) [inline lib/errors.js]: Not sure it's a good idea ;-(

**Comment by zxqfox** [comment]
:+1: at all.

File: lib/errors.js (hunk 2 above)
Comment (markelog) [inline lib/errors.js]: Meaning?

File: lib/cli.js (hunk 1 above)
Comment (markelog) [inline lib/cli.js]: Could you elaborate?

**Comment by markelog** [comment]
> Is it breaking change?

No

File: lib/cli.js (hunk 1 above)
Comment (zxqfox) [inline lib/cli.js]: Didn't see where you readd it to checker. Suggesting that it wasn't used in code.

File: lib/errors.js (hunk 2 above)
Comment (zxqfox) [inline lib/errors.js]: Just thinking about what Errors object should do. Reporters can format this kind of errors more wisely.
Nvm, it's another story.

**Comment by mikesherov** [comment]
:+1:

**Pull request closed without merging**